lua/entities/emplacements_turret_base/shared.lua

```diff
@@ -67,73 +95,70 @@ function ENT:Use( plr )
         if call == false then return end
 
         if IsValid( plr.CurrentEmplacement ) then
-            if SERVER then
-                -- plays sound on self to remind players this is an intended feature
-                self:EmitSound( "common/wpn_denyselect.wav", 60 )
-            end
-
-            return
+            plr.CurrentEmplacement:EmplacementDisconnect() -- hotswap emplacements! feels much better than being denied
+            
         end
-
-        self:SetShooter( plr )
-        self:StartShooting()
-        self.ShooterLast = plr
+        
+        self:EmplacementConnect( plr )
+        
     else
         if plr == self.Shooter then
-            self:SetShooter( nil )
-            self:FinishShooting()
+            self:EmplacementDisconnect()
         end
     end
 end
 
 function ENT:Think()
-    if not IsValid( self.turretBase ) and SERVER then
-        SafeRemoveEntity( self )
+    if not IsValid( self.turretBase ) then
+        if SERVER then SafeRemoveEntity( self ) end
+        return nil
```

Comment:
I still think it's a good idea to get rid of this logic entirely, as outlined here: https://github.com/CFC-Servers/emplacements/pull/10#discussion_r702390322

When you create the `turretBase`, you can use `turretBase:DeleteOnRemove( self )` or something like that. Then you won't need to do these checks on` :Think`

ref:
https://wiki.facepunch.com/gmod/Entity:DeleteOnRemove

Reply:
That's a function? good to know.
It's not a huge impact imo if this inefficient bit of code stays here for now but DeleteOnRemove would be much better.